frappe_private_comment/public/js/controls/replies.js

```diff
@@ -0,0 +1,290 @@
+      .on("click", () => handle_reply_edit(commentSelector, "#comment-" + reply.name));
+
+    moreButton.append(dropdownMenu);
+    actionButtons.append(editButton, replyButton, moreButton);
+    $replyContent.find(".text-muted").append(actionButtons);
+  });
+
+  $comment.after($replyContainer);
+}
+
+function addThreadedReply(commentSelector, doctype) {
+  const $comment = $(commentSelector);
+  const commentId = $comment.data("name");
+
+  frappe.call({
+    method: "frappe_private_comment.overrides.whitelist.comment.get_comment_replies",
+    args: {
+      reference_doctype: doctype,
+      comment_id: commentId,
+    },
+    callback: (res) => {
+      if (res.exc) {
+        console.error(res.exc);
+        return;
+      }
+
+      if (!res.message || !res.message.length) {
+        // No replies found
+        return;
+      }
+
+      render_replies(commentSelector, res.message);
+    },
+  });
+}
+
+function handle_reply(time_line_item) {
+  const commentBox = $(time_line_item).find(".comment-edit-box");
+
+  if (!commentBox.length) {
+    console.error("Comment box not found");
+    return;
+  }
+
+  commentBox.css("display", "block");
+  commentBox.empty();
+
+  const replyEditBox = $("<div>").addClass("reply-edit-box").html(`
+      <div class="ql-container ql-snow">
+        <div class="ql-editor" contenteditable="true" style="max-height: 40px;">
+        </div>
+      </div>
+      <div class="reply-actions" style="background-color: white; padding: 4px;">
+        <button class="btn btn-sm btn-primary submit-reply">Submit</button>
+        <button class="btn btn-sm btn-default cancel-reply">Cancel</button>
+      </div>
+    `);
+
+  commentBox.append(replyEditBox);
+
+  replyEditBox.find(".ql-editor").focus();
+
+  replyEditBox.find(".submit-reply").on("click", () => {
+    const replyContent = replyEditBox.find(".ql-editor").html();
+    submit_reply(time_line_item, replyContent);
+  });
+
+  replyEditBox.find(".cancel-reply").on("click", () => {
+    commentBox.empty().css("display", "none");
+  });
+
+  // Scroll to the comment box
+  $("html, body").animate(
+    {
+      scrollTop: commentBox.offset().top - $(window).height() / 2,
+    },
+    1000
+  );
+}
+
+function submit_reply(time_line_item, content) {
+  frappe.call({
+    method: "frappe.desk.form.utils.add_comment",
+    args: {
+      reference_doctype: this.cur_frm.doctype,
+      reference_name: this.cur_frm.docname,
+      custom_reply_to: $(time_line_item).data("name") || null,
+      content: content,
+      comment_email: frappe.session.user,
+      comment_by: frappe.session.user_fullname,
+    },
+    callback: (r) => {
+      if (r.message) {
+        $(time_line_item).find(".comment-edit-box").empty().css("display", "none");
+        frappe.utils.play_sound("click");
+        update_comments_timeline();
+        addThreadedReply(time_line_item, this.cur_frm.doctype);
+      }
+    },
+  });
+}
+
+function handle_reply_copy(commentSelector) {
```

Comment:
Use case?

Reply:
Had to write this function to add the 'Copy Link' feature of comments.

Reply:
Can remove the feature altogether if required.